### PR TITLE
[positional encoding] Implement positional encoding layer

### PR DESCRIPTION
**Pull request by lhs8928**
Commit1:     [positional encoding] implement positional encoding layer
    
     - Positional encoding just needed to be calculated only once
       so make its lifespan as max lifespan
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commi2:     [unittest] generate positional encoding unittest
    
     - Generate positional encoding layer/model unittest.
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: nntrainer/layers/positional_encoding_layer.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  input.add(pe_partial, output);
+}
+
+void PositionalEncodingLayer::calcDerivative(RunLayerContext &context) {
+  const nntrainer::Tensor &incoming_derivative =
+    context.getIncomingDerivative(SINGLE_INOUT_IDX);
+  nntrainer::Tensor &outgoing_derivative =
+    context.getOutgoingDerivative(SINGLE_INOUT_IDX);
+
+  outgoing_derivative.copyData(incoming_derivative);
```
Comment (jijoongmoon) [inline nntrainer/layers/positional_encoding_layer.cpp:78]: This Positional Encoding Layer is not trainable. then.

File: nntrainer/layers/positional_encoding_layer.cpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  unsigned int max_token_size =
+    std::get<props::MaxTimestep>(positional_encoding_props);
+
+  unsigned int model_dim = context.getInput(SINGLE_INOUT_IDX).getDim().width();
+
+  nntrainer::Tensor &pe = context.getTensor(
+    weight_idx[PositionalEncodingParams::positional_encoding]);
+
+  float value;
+  for (unsigned int i = 0; i < max_token_size; ++i) {
```
Comment (jijoongmoon) [inline nntrainer/layers/positional_encoding_layer.cpp:92]: It might be better to use std::transform to accelerate.

**Review by jijoongmoon — APPROVED** [review]
overall, LGTM

**Review by songgot — APPROVED** [review]
LGTM

File: nntrainer/layers/positional_encoding_layer.cpp (hunk 1 above)
Comment (lhs8928) [reply nntrainer/layers/positional_encoding_layer.cpp:78]: Exactly

File: nntrainer/layers/positional_encoding_layer.cpp (hunk 2 above)
Comment (lhs8928) [reply nntrainer/layers/positional_encoding_layer.cpp:92]: To calculate positional encoding value, it needs position, index information which is i, j in this for loop.
However in std::transform there is no way to get this information.
Please correct me if I'm wrong.

**Review by jihochu — APPROVED** [review]
LGTM

**Pull request merged**